### ISSUE TITLE
Akka.Streams.Util.IteratorAdapter destroys stack trace of captured errors.

### ISSUE THREAD

**Issue by gambrose** [akkadotnet/akka.net#3096]
`Akka.Streams.Util.IteratorAdapter` captures exceptions when calling `HasNext` which are then re-thrown. When it re-throws the exception the original stack trace of what caused the exception is lost. Instead replaced with a stack trace that just leads to `Akka.Streams.Util.IteratorAdapter.Next()`.
It would be more useful if the captured exception was wrapped (possibly using an `AggregateException`) so that the original stack trace could be preserved.

- Which Akka.Net version you are using
1.2
- On which platform you are using Akka.Net
net462

**Comment by robertlyson** [comment]
I tried to reproduce this on recent `dev` branch but seems to be fine. Maybe I'm missing something?

```
public class IteratorAdapterSpec
{
    [Fact]
    public void IteratorAdapter_original_exception_should_be_preserved()
    {
        var iteratorAdapter = new IteratorAdapter<object>(new ThrowExceptioEnumerator<object>());

        iteratorAdapter.Next();
    }

    class ThrowExceptioEnumerator<T> : IEnumerator<T>
    {
        public T Current => throw new NotImplementedException();

        object IEnumerator.Current => throw new NotImplementedException();

        public void Dispose()
        {
            throw new NotImplementedException();
        }

        public bool MoveNext()
        {
            throw new NullReferenceException("ups");
        }

        public void Reset()
        {
            throw new NotImplementedException();
        }
    }
}
```

Output:
```
System.NullReferenceException : ups
   at Akka.Streams.Util.IteratorAdapter`1.Next()
   at Akka.Streams.Tests.Util.IteratorAdapterSpec.IteratorAdapter_original_exception_should_be_preserved() in C:\dev\projects\akka.net\src\core\Akka.Streams.Tests\Util\IteratorAdapterSpec.cs:line 17
```

**Comment by marcpiechura** [comment]
@robertlyson I think the problem is that there should be another line after `1.Next()` which points to `MoveNext` because that’s the actual place where the exception is thrown, not `Next()`

**Comment by robertlyson** [comment]
Ah, got it :+1:

**Comment by marcpiechura** [comment]
Fixed in #3343
